### DISCUSSION TITLE
Copilot: This extension is disabled because the installed version is incompatible with Visual Studio.

### DISCUSSION THREAD

**Discussion by axelriet** [community/community#118503] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

<img width="317" alt="image" src="https://github.com/community/community/assets/7425484/fef152a8-a3ed-4c41-acb9-7499538448b6">

<img width="794" alt="image" src="https://github.com/community/community/assets/7425484/1cc3eb5b-2cb3-4dde-b424-78a19e1ab08e">

<img width="257" alt="image" src="https://github.com/community/community/assets/7425484/554f6dd6-15d0-47b6-bdbc-d352acc409d3">

Cannot uninstall:

<img width="349" alt="image" src="https://github.com/community/community/assets/7425484/5085b820-a77f-4d3c-b9e1-b6a2d6975615">

09-Apr-24 23:41:52 - Initializing Uninstall...
09-Apr-24 23:41:52 - Extension Details...
09-Apr-24 23:41:52 - 	Identifier         : GitHub.Copilot
09-Apr-24 23:41:52 - 	Name               : GitHub Copilot
09-Apr-24 23:41:52 - 	Author             : GitHub
09-Apr-24 23:41:52 - 	Version            : 1.147.0.0
09-Apr-24 23:41:52 - 	Description        : GitHub Copilot is an AI pair programmer that helps you write code faster and with less work. Requires Visual Studio 2022 17.5.5 or later.
09-Apr-24 23:41:52 - 	Locale             : en-US
09-Apr-24 23:41:52 - 	MoreInfoURL        : 
09-Apr-24 23:41:52 - 	InstalledByMSI     : False
09-Apr-24 23:41:52 - 	SupportedFrameworkVersionRange : [0.0,2147483647.2147483647]
09-Apr-24 23:41:52 - 	Supported Products : 
09-Apr-24 23:41:52 - 		Microsoft.VisualStudio.Community
09-Apr-24 23:41:52 - 			Version : [17.5.33627,18.0)
09-Apr-24 23:41:52 - 			ProductArchitecture : amd64
09-Apr-24 23:41:52 - 		Microsoft.VisualStudio.Community
09-Apr-24 23:41:52 - 			Version : [17.5.33627,18.0)
09-Apr-24 23:41:52 - 			ProductArchitecture : arm64
09-Apr-24 23:41:52 - 	References         : 
09-Apr-24 23:41:52 - 	Prerequisites      : 
09-Apr-24 23:41:52 - 		-------------------------------------------------------
09-Apr-24 23:41:52 - 		Identifier   : Microsoft.VisualStudio.Component.IntelliCode
09-Apr-24 23:41:52 - 		Name         : IntelliCode
09-Apr-24 23:41:52 - 		Version      : [17.4.33213,18.0)
09-Apr-24 23:41:55 - Blocking processes:
09-Apr-24 23:41:55 - InstallProgressPage ChangeType: 'SingleTarget' operationCount: '2' Text: 'Modifying Microsoft Visual Studio Community 2022...'
09-Apr-24 23:41:55 - The following target products have been selected...
09-Apr-24 23:41:55 - 	Microsoft Visual Studio Community 2022
09-Apr-24 23:41:55 - Uninstalling 'GitHub Copilot', version 1.147.0.0.
09-Apr-24 23:41:55 - pkgCatalogPath is 'C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\EXTENSIONS\GITHUB.COPILOT\catalog.json'.
09-Apr-24 23:41:55 - Instantiate download manager silently: False
09-Apr-24 23:41:57 - Windows Installer found the package 'Microsoft.VisualStudio.Community.Msi,version=17.10.34705.215,machinearch=arm64' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Microsoft.Azure.PowerShell,version=17.10.34705.114' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Microsoft.Azure.PowerShellNew,version=17.10.34705.114,chip=x86' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Microsoft.VisualCpp.RuntimeDebug.14.ARM64,version=14.40.33617,chip=x64,machinearch=arm64' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Maui.Sdk.net7.7.0.101,version=7.0.101.0,machinearch=x64' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Maui.Resizetizer.Sdk.7.0.101,version=7.0.101.0,machinearch=x64' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Maui.Templates.net7.7.0.101,version=7.0.101.0,machinearch=x64' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Microsoft.NetCore.TargetingPack.5.0.5.0.0,version=17.10.34705.114,chip=x64' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Microsoft.Azure.PowerShell,version=17.10.34705.114' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Maui.Sdk.net7.7.0.101,version=7.0.101.0,machinearch=x64' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Maui.Resizetizer.Sdk.7.0.101,version=7.0.101.0,machinearch=x64' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Maui.Templates.net7.7.0.101,version=7.0.101.0,machinearch=x64' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Microsoft.NetCore.TargetingPack.5.0.5.0.0,version=17.10.34705.114,chip=x64' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Microsoft.Azure.PowerShell,version=17.10.34705.114' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Maui.Sdk.net7.7.0.101,version=7.0.101.0,machinearch=x64' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Maui.Resizetizer.Sdk.7.0.101,version=7.0.101.0,machinearch=x64' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Maui.Templates.net7.7.0.101,version=7.0.101.0,machinearch=x64' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:57 - Windows Installer found the package 'Microsoft.NetCore.TargetingPack.5.0.5.0.0,version=17.10.34705.114,chip=x64' superseded but corresponding compatible package information was not found. Attempts to repair will be skipped and that to uninstall may leave the cache behind.
09-Apr-24 23:41:58 - vsPkg.Version is '1.147.0.0'.  extension.Header.Version is '1.147.0.0'
09-Apr-24 23:41:59 - System.InvalidOperationException: Uninstall of GitHub Copilot failed. To uninstall this extension please uninstall all of the following components: GitHub Copilot
   at Microsoft.VisualStudio.ExtensionManager.UninstallModule.PerformSetupEngineUnInstall(IInstalledExtension extension, InstallFlags installFlags, String& downgradePackageInstallPath, Boolean downgrade)
   at Microsoft.VisualStudio.ExtensionManager.UninstallModule.UninstallInternal(IInstalledExtension extension, Boolean forceDelete, Version targetedVSVersion, InstallFlags installFlags, Boolean downgrade)
   at Microsoft.VisualStudio.ExtensionManager.UninstallModule.Uninstall(IInstalledExtension extension, InstallFlags installFlags, Version targetedVSVersion, Boolean forceDelete)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.Uninstall(IInstalledExtension extension, InstallFlags installFlags, Version targetedVSVersion)
   at VSIXInstaller.InstallProgressPage.UninstallVsix(String vsixId, SupportedVSSKU targetSku, IProgress`1 progress)

The Updates tab does not show any update. The Installed tab shows CoPilot and Copilot Chat as "incompatible". I could inunstall Copilot Chat but only "Revert" CoPilot, and not uninstall it. After reverting I could update, but the issue persisted.

Paid subscription running:

<img width="707" alt="image" src="https://github.com/community/community/assets/7425484/df947db5-abed-4654-a1fa-6cfaeba73423">

**Comment by Edward-Mendez-at-hciq** [comment]
I had this same error message popup this morning using VS 2022 Community17.10.0 Preview 2. If I went to my extensions for GitHub Copilot it said it was incompatible. 
I also have VS 2022 Community 17.9.5 installed with GitHub Copilot and that works and the Extension doesn't said it is incompatible. inside my VS 17.9.5 I removed the two extensions GitHub Copilot and GitHub Copilot Chat
I also updated VS preview to 17.10.0 Preview 3.

The error went away. Although I don't have GitHub Copilot installed in 17.9.5 anymore, I can live with that.

**Reply by ayim** [reply, marked as answer]
Hi @Edward-Mendez-at-hciq,

Yes, this was a bug in 17.10 Preview 2 and was fixed for Preview 3.
Please let us know if you see any other issues with GitHub Copilot in Visual Studio 2022.

-- 
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio

**Comment by HiveTyrant** [comment]
Seems like it is now a built in feature of VS-2022 preview according to this article:
https://devblogs.microsoft.com/visualstudio/introducing-the-new-copilot-experience-in-visual-studio/

**Comment by fubaduba** [comment]
Hey @axelriet were you able to resolve this?

**Comment by beefydog** [comment]
Anyone figure out how to solve? It's a bit annoying since I don't use copilot (which is somewhat useless)

**Reply by ayim** [reply]
Hi @beefydog, it sounds like you've encountered this issue: https://developercommunity.visualstudio.com/t/Visual-Studio-in-a-startup-loop-because-/10662828

Although we released a fix in VS 17.10.1, the recommended workaround is as follows:

> run the following command under a Developer Powershell instance to remove the old extension from all Visual Studio instance:
`vsixinstaller /uninstall:"VS.GitHub.Copilot"`

You'll still have GitHub Copilot's component in the installer providing functionality, this just removes the old extension that causes the incompatible extension warning.

For some users, the `repair` functionality in the Visual Studio Installer has also been helpful.
![image](https://github.com/user-attachments/assets/6f5f6cf7-f20e-4045-8e52-0d2ab5fbbde2)
